pkg/query/endpointset_test.go

```diff
@@ -269,13 +271,35 @@ func (e *testEndpoints) CloseOne(addr string) {
 	delete(e.srvs, addr)
 }
 
+func truncateAndEscapeQuotes(s string) string {
+	// Truncate string.
+	if len(s) > externalLabelLimit {
```

Comment:
Ok, this is an anti pattern. You don't want to use exactly the same code for doing things and asserting things. This is because if this code is broken, we don't know as assertion would follow. I would have large string in `expectedOutput` variable (: